common/common-rest/src/main/java/org/apache/servicecomb/common/rest/codec/produce/ProduceProcessorManager.java

```diff
@@ -17,24 +17,43 @@
 
 package org.apache.servicecomb.common.rest.codec.produce;
 
+import java.util.ServiceLoader;
+
 import javax.ws.rs.core.MediaType;
 
 import org.apache.servicecomb.foundation.common.RegisterManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public final class ProduceProcessorManager extends RegisterManager<String, ProduceProcessor> {
+  private static final Logger LOGGER = LoggerFactory.getLogger(ProduceProcessorManager.class);
+
+  private static final ServiceLoader<ProduceProcessor> produceProcessor = ServiceLoader.load(ProduceProcessor.class);
+
   private static final String NAME = "produce processor mgr";
 
   public static final String DEFAULT_TYPE = MediaType.APPLICATION_JSON;
 
   public static final ProduceProcessorManager INSTANCE = new ProduceProcessorManager();
 
-  public static final ProduceProcessor JSON_PROCESSOR = new ProduceJsonProcessor();
+  public static final ProduceProcessor JSON_PROCESSOR = chooseProduceProcessor(MediaType.APPLICATION_JSON);
 
-  public static final ProduceProcessor PLAIN_PROCESSOR = new ProduceTextPlainProcessor();
+  public static final ProduceProcessor PLAIN_PROCESSOR = chooseProduceProcessor(MediaType.TEXT_PLAIN);
 
   public static final ProduceProcessor DEFAULT_PROCESSOR = JSON_PROCESSOR;
 
   private ProduceProcessorManager() {
     super(NAME);
+    produceProcessor.forEach(processor -> register(processor.getName(), processor));
+  }
+
+  public static ProduceProcessor chooseProduceProcessor(String type) {
+    for (ProduceProcessor processor : produceProcessor) {
+      if (type.equals(processor.getName()))
+        return processor;
+    }
+    LOGGER.warn("Getting the current produceProcessor type {} failed, use default produceProcessor: application/json",
+        type);
+    return new ProduceJsonProcessor();
```

Comment:
This is not fixed as mentioned in previous comment. Do we need to create a new instance or can reuse JSON_PROCESSOR ?

Reply:
now use SPIServiceUtils,if not get, it will return null
![image](https://user-images.githubusercontent.com/30716999/41077353-c266d6d6-6a48-11e8-8324-c8c2a6a3f62e.png)